gobblin-compaction/src/main/java/org/apache/gobblin/compaction/action/CompactionCompleteFileOperationAction.java

```diff
@@ -73,6 +74,11 @@ public CompactionCompleteFileOperationAction (State state, CompactionJobConfigur
    * Create a record count file containing the number of records that have been processed .
    */
   public void onCompactionJobComplete (FileSystemDataset dataset) throws IOException {
+    if (dataset instanceof SimpleFileSystemDataset
```

Comment:
How about extending the FileSystemDataset interface with an isVirtual() method that defaults to false to avoid this specific instanceof and casting?

Reply:
+1